### ISSUE TITLE
User can see preview of Forum Threads where he has no access

### ISSUE THREAD

**Issue by Bjoern3003** [PHPfox-Official/phpfox-v4-issues#2864]
In the Latest Thread Block, User can also see Preview Threads (The Headline) of Categories where he is not able to access.

**Comment by Bjoern3003** [comment]
Hotifx:

in File PF.Site/Apps/core-forums/Service/Post/Post.php

```php
public function getRecentForForum($forumId = 0, $iLimit = 4)
    {
        $aWhere = [
            'ft.group_id' => 0,
            'ft.view_id' => 0
        ];
        if ($forumId) {
            $aWhere['ft.forum_id'] = $forumId;
        }

        $sOrder = 'fp.time_stamp DESC, ft.time_update DESC';

        $this->database()->select('fp.*')
            ->from($this->_sTable, 'fp')
            ->join(':forum_thread', 'ft', 'ft.thread_id = fp.thread_id')
            ->where($aWhere)
            ->order($sOrder)
            ->limit(0, $iLimit);

        $this->database()->union()->unionFrom('fp');

        $posts = $this->database()->select('fp.*, fpt.text_parsed, ft.title as thread_title, ' . Phpfox::getUserField())
            ->join(':forum_thread', 'ft', 'ft.thread_id = fp.thread_id')
            ->join(':forum_post_text', 'fpt', 'fpt.post_id = fp.post_id')
            ->join(':user', 'u', 'u.user_id = fp.user_id')
            ->order($sOrder)
            ->group('ft.thread_id');

        if(!Phpfox::isAdmin())
            $posts
                ->join(Phpfox::getT('forum_access'), 'fa', 'fa.forum_id = ft.forum_id')
                ->join(Phpfox::getT('forum_moderator'), 'fm', 'fm.forum_id = ft.forum_id')
                ->where('(fa.user_group_id = ' . Phpfox::getUserBy('user_group_id') . ' AND fa.var_value = 1 AND fa.var_name = "can_view_thread_content") OR (fm.user_id = '.Phpfox::getUserId().')');

        return $posts->executeRows();
    }
```

**Comment by harrison05** [comment]
Hi @Bjoern3003 

We will check this issue and your solution.

Thanks.

**Comment by PhpFoxJohnJr** [comment]
Again...forum was made into an app so bugs could be fixed faster and on January 29 it was confirmed as a bug?

**Comment by harrison05** [comment]
Fixed on Forums v4.6.4
